### ISSUE TITLE
Organize + document `certbot/tests/testdata` directory

### ISSUE THREAD

**Issue by ynasser** [certbot/certbot#4917]
We have a horribly organized `certbot/tests/testdata/` directory, and it's not clear which of our testing certificates go with which keys, and this has caused some confusion. Ideally, this would be documented somewhere around here: https://certbot.eff.org/docs/contributing.html#testing.

Here's a small sample of what's involved in part of this (e.g. checking that a private key goes with a certificate):

```
$ openssl x509 -noout -modulus -in cert.pem | openssl md5
(stdin)= c1cc74880b0f3366db7c9fc48a359dff
$ openssl rsa -noout -modulus -in rsa256_key.pem | openssl md5
(stdin)= f7f1dda5f208c1718c3c18189114ddd6
```

Since the hashes don't match, they don't go together (but those two are used together in a lot of the unit tests).

Relevant preliminary reading on the differences between `fullchain`, `cert`, and `chain` files for whoever takes this on: https://certbot.eff.org/docs/using.html#where-are-my-certificates

**Comment by Baeumla** [comment]
Since I have already worked on this for my last PR, I will take care of this issue. I will have to take a closer look into the `_test` files so I don't know when I will be able to create the PR for this. Will do it as soon as possible.

@ynasser you can assign this to me if possible.

**Comment by ynasser** [comment]
Oh, wow thanks @Baeumla! I'll just assign it to myself so we know who's managing it, since GH won't let me assign people not in the organization.

**Comment by Baeumla** [comment]
BTW: When I was generating the Sphinx documentation on my macOS with the `make -C docs clean html man` command, I was having import errors. I had to manually install `repoze.sphinx.autointerface` and `sphinx_rtd_theme`. I think this should be mentioned in the Developer Guide. 

To install those packages I used the pip installer.
```
pip install repoze.sphinx.autointerface
pip install sphinx_rtd_theme 
```

**Comment by ynasser** [comment]
Thanks for pointing that out @Baeumla. I've made the following issue to address this: #4946.

**Comment by Baeumla** [comment]
Okay. @ynasser do you actually know why there is a second testdata directory `certbot/acme/acme/testdata`? This one has a README on how the certificates and keys got created. Wouldn't it be better if there would be only one testdata-directory, or is it better to separate the acme from the certbot testdata?

I checked some of the keys and certs in borth testdata directories and for example the `rsa256_key.pem,` `rsa512_key.pem`, and `cert.pem` are basically the same files. Here is the complete diff of the two directories: https://gist.github.com/Baeumla/c08c97d2eef3cc4c0e495413e225a100

**Comment by ynasser** [comment]
Yes, we've intentionally kept references to certbot out of acme because the protocol-specific code shouldn't rely on the client-specific code.

There will end up being duplicates between `acme/acme/testdata` and `certbot/certbot/tests/testdata`, but that is OK. Including a README in both would be a good idea.

**Comment by Baeumla** [comment]
@ynasser Okay, thank you for this information. I'm working on this for a week now and I realized that it's a little bit more of a mess than I initially thought. There are files in this folder which are never used, e.g. `cert.b64jose` or `dsa_cert.pem`. However, I will try to finish up my work over the weekend and create the PR for this.

**Comment by ynasser** [comment]
@Baeumla No worries, take your time! And sorry for how much of a mess it is :(

**Comment by Baeumla** [comment]
@ynasser I came across another thing I wasn't really sure about: In the `main_test.py` the `rsa512_key_2.pem` gets loaded as JSON Web Key while in fact it is actually a regular PEM key here: https://github.com/certbot/certbot/blob/master/certbot/tests/main_test.py#L39 . 

I converted the `rsa512_key.pem` into a JWK and tried to exchange it with the `rsa512_key_2.pem` in the `main_test.py` but the test raised some errors.. So I was wondering if this was intended? Unfortunately, I haven't found out yet how to load the "real" JWK for the test. Here is how I generated the JWK and how it looks: https://gist.github.com/Baeumla/6ac442fa56c2f7a3c77806ac589cb6c3

**Comment by ynasser** [comment]
@Baeumla

I think that with a name like `JWKRSA.load`, you'd expect it to be loading a JWK, but it's name is actually misleading. From the `load` function's [docstring](https://github.com/certbot/certbot/blob/master/acme/acme/jose/jwk.py#L96), the first parameter is a "Public or private key serialized as PEM or DER". So, `load` actually returns a JWK, given a PEM or a DER :)

**Comment by ynasser** [comment]
Closed via #4983
